log-manager/src/main/java/io/airlift/log/LoggingConfiguration.java

```diff
@@ -16,19 +16,28 @@
 package io.airlift.log;
 
 import io.airlift.configuration.Config;
+import io.airlift.configuration.ConfigDescription;
+import io.airlift.configuration.DefunctConfig;
 import io.airlift.configuration.LegacyConfig;
+import io.airlift.log.RollingFileHandler.CompressionType;
 import io.airlift.units.DataSize;
 
-import static io.airlift.units.DataSize.Unit.BYTE;
+import javax.validation.constraints.NotNull;
+
+import static io.airlift.units.DataSize.Unit.GIGABYTE;
 import static io.airlift.units.DataSize.Unit.MEGABYTE;
 
+@DefunctConfig("log.max-size-in-bytes")
 public class LoggingConfiguration
 {
     private boolean consoleEnabled = true;
     private String logPath;
     private DataSize maxSize = new DataSize(100, MEGABYTE);
+    private DataSize maxTotalSize = new DataSize(1, GIGABYTE);
```

Comment:
for maxSize 100M and maxTotalSize 1G, shouldn't maxHistory be initialized to 10?

if the rollover is time and/or size -based (whatever triggers first), maybe just add this as a comment

Reply:
maxHistory is not enforced for the new logger.  It is only present for the legacy logger.

Reply:
We could split the config and bind conditionally based on legacy/new, but that's probably overkill.

Reply:
Yes, I think it is overkill.  In a few months we can remove the old logging system and the config.